### DISCUSSION TITLE
Create a Data Provider from a data provider transformation

### DISCUSSION THREAD

**Discussion by Gomeyy** [Budibase/budibase#4050] [Ideas]
I am currently in a situation where I need to call the same api twice in order to make use of charts with part of the data from my rest source and since charts all need different formats, I was wondering if it was possible to create a data provider as a transformer from another data provider so you don't call the api twice and are able to get your data in a different format. That or the possibility of creating custom inputs on charts would help a ton.
Thank you,
Appreciate all the work put into this!

**Comment by mjashanks** [comment]
Hi @Gomeyy - this is a great Idea, but is not possible at the moment.  Converting this to an Idea, so that it gets some consideration in the future. Thank you!

**Reply by Gomeyy** [reply]
Hope it gets considered, thank you!

**Reply by joebudi** [reply]
Hi @Gomeyy . I'm the co-founder of Budibase. Would you be interested in jumping on a quick call to discuss how you are using Budibase, and any feedback you may have?
https://calendly.com/budibase/joe

**Reply by Gomeyy** [reply]
@joebudi Would love too, setup a meeting for tomorrow at 11, talk to you then

**Comment by nntrongnghia** [comment]
Hi @Gomeyy, I ran into the same issue as your. Do you have any idea to avoid running the same API twice?
